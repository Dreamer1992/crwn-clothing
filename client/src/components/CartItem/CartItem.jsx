import { memo } from 'react';

import {
  CartItemContainer,
  CartItemImage,
  ItemDetailsContainer,
} from './CartItem.styles';

const CartItem = ({ item: { imageUrl, price, name, quantity } }) => (
  <CartItemContainer>
    <CartItemImage src={imageUrl} alt="item" />

    <ItemDetailsContainer>
      <span className="name">{name}</span>
      <span className="price">
        {quantity} x ${price}
      </span>
    </ItemDetailsContainer>
  </CartItemContainer>
);

export default memo(CartItem);
