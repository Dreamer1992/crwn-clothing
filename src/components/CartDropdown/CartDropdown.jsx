import { Button } from '../UI';
import './CartDropdown.styles.scss';

const CartDropdown = () => {
  return (
    <div className="cart-dropdown">
      <div className="cart-items"></div>
      <Button>GO TO CHECKOUT</Button>
    </div>
  );
};

export default CartDropdown;
