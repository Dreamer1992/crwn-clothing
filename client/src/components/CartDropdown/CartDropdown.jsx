import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { selectCartItems } from '../../redux/cart/cartSelectors';
import CartItem from '../CartItem/CartItem';

import { toggleCartHidden } from '../../redux/cart/cartActions';

import {
  CartDropdownContainer,
  CartItemsContainer,
  EmptyMessageContainer,
  CartDropdownButton,
} from './CartDropdown.styles';

export const CartDropdown = ({ cartItems, history, dispatch }) => {
  // const history = useHistory();
  // const dispatch = useDispatch();

  // const cartItems = useSelector(selectCartItems);

  return (
    <CartDropdownContainer>
      <CartItemsContainer>
        {cartItems.length ? (
          cartItems.map((cartItem) => (
            <CartItem key={cartItem.id} item={cartItem} />
          ))
        ) : (
          <EmptyMessageContainer>Your cart is empty</EmptyMessageContainer>
        )}
      </CartItemsContainer>

      <CartDropdownButton
        onClick={() => {
          history.push('/checkout');
          dispatch(toggleCartHidden());
        }}
      >
        GO TO CHECKOUT
      </CartDropdownButton>
    </CartDropdownContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
});

export default withRouter(connect(mapStateToProps)(CartDropdown));
