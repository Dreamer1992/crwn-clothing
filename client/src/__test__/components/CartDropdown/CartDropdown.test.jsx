import { shallow } from 'enzyme';

import CartDropdown from '../../../components/CartDropdown/CartDropdown';
import CartItem from '../../../components/CartItem/CartItem';

import { Provider } from 'react-redux';

import { toggleCartHidden } from '../../../redux/cart/cartActions';

describe('CartDropdown component', () => {
  let wrapper;
  let mockHistory;
  let mockDispatch;
  const mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

  beforeEach(() => {
    mockHistory = { push: jest.fn() };
    mockDispatch = jest.fn();

    const mockProps = {
      cartItems: mockCartItems,
      history: mockHistory,
      dispatch: mockDispatch,
    };

    wrapper = shallow(<CartDropdown {...mockProps} />);
  });

  it('should render CartDropdown component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
