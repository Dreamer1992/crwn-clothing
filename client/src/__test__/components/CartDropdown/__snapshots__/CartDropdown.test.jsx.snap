// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component should render CartDropdown component 1`] = `ShallowWrapper {}`;

exports[`should render CartDropdown component 1`] = `<CartDropdown />`;
